test/WebJobs.Script.Tests/FunctionsSyncServiceTests.cs

```diff
@@ -78,6 +85,22 @@ public async Task StartAsync_TokenCancelledBeforeTimeout_DoesNotSyncTriggers()
             _mockSyncManager.Verify(p => p.TrySyncTriggersAsync(true), Times.Never);
         }
 
+        [Theory]
```

Comment:
In addition to this test, we also have full standby mode E2E tests. I could see in those tests that before my fix we would start the timer for a standby host instance. Now we only do it when NOT in standby mode.

Moving from standby mode to specialized causes a new FunctionSyncService to be created/started (the service lifetime is scoped to host). For a placeholder host it noops.